src/block-management/block-holder.js

```diff
@@ -75,6 +82,14 @@ export class BlockHolder extends React.Component<PropsType, StateType> {
 		this.props.onSelect( this.props.clientId );
 	};
 
+	onRemoveBlockCheckUpload = ( mediaId: number ) => {
+		if ( hasAction( 'blocks.onRemoveBlockCheckUpload' ) ) {
+			// now remove the action as it's  a one-shot use and won't be needed anymore
+			removeAction( 'blocks.onRemoveBlockCheckUpload', 'gutenberg-mobile/blocks' );
```

Comment:
Maybe it would be a good option to put those two arguments in some constants so that we can reuse them in other parts of the code?

Reply:
I was looking into the Gutenberg code for other `addAction` / `removeAction` calls [elsewhere](https://github.com/WordPress/gutenberg/blob/daa2096d24ded61388b439feebe1593d5e10ca80/packages/editor/src/components/post-locked-modal/index.js#L32-L46) and they seem to have the text right there so, followed that. Does that make sense?

Adding a note here as well - I also considered using an instance ID as seen in that piece of code before to make sure to have different actions for each of the blocks, but given it all happens in a brief amount of time (from the time the user presses the trash icon making the action be added, and then when the component gets unmounted where the action gets removed), it didn't make sense to add the complexity of having to add a props to pass down to the component (which essentially would make it re-render and then again, make the bridge change unnecessary at the block holder level).

Reply:
Well if someone tries to change action name, he would need to change it on 3 different places. I think that it's a good practice to have one constant which will hold the name of the action?

Reply:
> I think that it's a good practice 

sure thing - don't get me wrong, I totally understand what you're saying in terms of how having a constant is useful to avoid having the text modified and introduce bugs this way 😅 👍. 

The thing (I left a [comment about it in the other PR](https://github.com/WordPress/gutenberg/pull/14391#discussion_r265973573)) is we would need to define it in `gutenberg-mobile` (where block holder lives) and import it in the gutenberg repo (where the Image component lives) making the block holder even aware of the Image component, which seems off. If we do it the other way around then we make the Image component aware of something defined in gutenberg-mobile block-holder. Both ways seem strange to me 🤔 . 

Perhaps I'm missing something, can you explain how would you do it? thanks in advance for your time and patience 🙇

Reply:
Ah ok thanks for the explanation, I thought that we are not on the same page regarding what should be done. Ok in that case :)

Reply:
Excellent! Glad it's sorted out 😄 - do you think this is good to be approved then @marecar3?